shared/src/utils/chainSwitching.ts

```diff
@@ -0,0 +1,35 @@
+import { providers } from "ethers";
+import { AVALANCHE_MAINNET_PARAMS } from "../constants/chainParameters";
+import { CHAINID } from "../constants/constants";
+
+export const switchChains = async (
+  provider: providers.Web3Provider,
+  chainId: number
+) => {
+  const hexChainId = "0x" + chainId.toString(16);
+
+  try {
+    await provider.provider.request!({
+      method: "wallet_switchEthereumChain",
+      params: [{ chainId: hexChainId }],
+    });
+  } catch (switchError: any) {
+    // This error code indicates that the chain has not been added to MetaMask.
+    if (switchError.code === 4902) {
```

Comment:
Ugh are these codes typed somewhere?  Probably not?

Reply:
Added some comments to clarify this